### PR TITLE
Add dev guide

### PR DESCRIPTION
**Pull request by nbrucy**
Add a developer contribution guide

Originally written by Corentin Cadiou
Co-authored-by: Corentin Cadiou <corentin.cadiou@fysik.lu.se>
Co-authored-by: Tine Colman <tinecolman@hotmail.com>

## PR Checklist

- [x] Pull request points to the `dev` branch.
- [x] Make sure that the doc will feature this new guide correctly

### CODE REVIEW

File: doc/dev_docs/developer_guide.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+
+Any member of the community can demand to demote a user to a lower role. Valid reasons to demote from administrator or maintainer include:
+- Absence of activity for a long period of time (e.g., no response to emails),
+- Repetitive mistakes in the management of the repositories (e.g., deleting important branches, merge without review, ignoring this guide).
+Demotion from the contributor list should only be done only in extreme cases (e.g., harassment, spamming) and will be discussed by the administrators on a case-by-case basis.
+
+Finally, the list of current administrators, maintainers, and contributors should be kept up-to-date in the CONTRIBUTORS.md file of the ramses-organisation repository. This list should be updated whenever a new user is promoted or demoted and include:
```
Comment (cphyc) [inline doc/dev_docs/developer_guide.md]: To quickstart the process, should we create the `CONTRIBUTORS.md` file and have a discussion here about who should be on the list?

Comment (nbrucy) [reply doc/dev_docs/developer_guide.md]: Good point. Maybe we should put the content of `doc/dev_docs/acknowledgements.md` inside it as well, no?

Comment (nbrucy) [reply doc/dev_docs/developer_guide.md]: So I replaced `doc/dev_docs/acknowledgements.md` by `doc/dev_docs/contributors.md`. Now its needs to be filled in :)

**Comment by nbrucy** [comment]
While we wait for the auto-doc building in the CI (issue #39), changes to the Read the docs website can be checked [here](https://ramses-fork-nbrucy.readthedocs.io/en/developper_guide/).

File: doc/dev_docs/developer_guide.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+Periodically, the developments on the `dev` branch that have proven to be robust, will be merged into the `stable` branch (see section 3.4). What is merged and what not, is decided through collegial discussions.
+At any point, any commit in `stable` should be a subset of the commits in `dev`.
+
+## Development workflow
+
+### Creating your working branch
+
+First, developers create their own private fork of the main repository. More information on forks and how to create them on github can be found [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo).
+
+Clone repo and Create branch TODO
```
Comment (tinecolman) [inline doc/dev_docs/developer_guide.md]: We forgot to remove a TODO here?

File: doc/dev_docs/developer_guide.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+- **Commit frequency**: We encourage frequent commits, but not too frequent. A good rule of thumb is to commit whenever a logical unit of work is completed and the code compiles and runs without errors.
+
+
+### Submitting a pull request
+
+Once you are ready to submit your changes, you can create a pull request (PR) on the GitHub interface. 
+Any new feature, bug fix, or improvement should be submitted as a PR to the `dev` branch.
+A PR should contain one logical unit of work. If you have multiple features or bug fixes, please submit them as separate PRs.
+More information on how to submit a PR on github, see:
+https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request
```
Comment (tinecolman) [inline doc/dev_docs/developer_guide.md]: Make this link clickable?

File: doc/dev_docs/developer_guide.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+It is the responsibility of the reviewer to ensure these conditions are met before merging the PR.
+
+We encourage the use of the GitHub interface to tag the PR with the appropriate labels (e.g., `enhancement`, `bug`, `on-merge: backport-to-stable`). The tag `on-merge: backport-to-stable` will automatically trigger backporting of this PR onto the `stable` branch once merged. We discourage manual backporting of bugfixes.
+
+### Updates to `stable`
+
+No pull-request nor commit should be pushed onto the `stable` branch, with the exception of:
+- (automated) backported commits,
+- regular updates of the `stable version.
```
Comment (tinecolman) [inline doc/dev_docs/developer_guide.md]: stable is missing a  ` for formatting

File: doc/dev_docs/developer_guide.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+In order to enforce this, the branch is protected (no one should commit directly onto it).
+
+**Backported commits** include *small* bugfixes and modification that should be included in the stable version (e.g., update of the contributor list). It excludes any breaking change, new feature, or experimental code. 
+
+**Updates of the stable version** happen on regular occasions (e.g., every Ramses User Meeting). The `stable` branch will be updated with the latest changes from `dev`. Collegial discussions will be held to decide which features are ready to be included in the `stable` version and which ones should be left out.
+
+
+## GitHub Management
+
+All developments related to the RAMSES community are hosted under the organisation is https://github.com/ramses-organisation.
```
Comment (tinecolman) [inline doc/dev_docs/developer_guide.md]: The word "is" to be removed.

**Review by tinecolman — COMMENTED** [review]
I've marked a few typos etc.

File: doc/dev_docs/developer_guide.md (hunk 4 above)
Comment (cphyc) [reply doc/dev_docs/developer_guide.md]:
```suggestion
- regular updates of the `stable` version.
```

File: doc/dev_docs/developer_guide.md (hunk 5 above)
Comment (cphyc) [reply doc/dev_docs/developer_guide.md]:
```suggestion
All developments related to the RAMSES community are hosted under the organisation https://github.com/ramses-organisation.
```

**Comment by rteyssier** [comment]
Good for me.

**Review by rteyssier — APPROVED** [review]

File: doc/dev_docs/developer_guide.md (hunk 3 above)
Comment (nbrucy) [reply doc/dev_docs/developer_guide.md]: done

File: doc/dev_docs/developer_guide.md (hunk 2 above)
Comment (nbrucy) [reply doc/dev_docs/developer_guide.md]: removed

**Review by blaizot — APPROVED** [review]
👍

**Pull request merged**